roles/job_runner/tasks/create_project.yml

```diff
@@ -1,16 +1,33 @@
 - name: Define a project, if error update AnsibleProject status then end play
   block:
-    - name: Launch project
+    - name: Launch project with credentials
       awx.awx.project:
         name: "{{ lookup('env','PROJECT_NAME') }}"
         description: "{{ lookup('env','PROJECT_DESCRIPTION') }}"
         organization: "{{ lookup('env','PROJECT_ORGANIZATION') }}"
         scm_update_on_launch: True
-        scm_type: git
-        scm_url: "{{ lookup('env','PROJECT_REPO') }}"
+        scm_type: "{{ lookup('env','SCM_TYPE') }}"
+        scm_url: "{{ lookup('env','PROJECT_REPO') | default(omit) }}"
         state: present
-        scm_branch: "{{ lookup('env','PROJECT_BRANCH') }}"
+        scm_branch: "{{ lookup('env','PROJECT_BRANCH'| default(omit)) }}"
       register: project
+      when:
+      - lookup('env','SCM_CREDENTIAL') is not defined
```

Comment:
@rooftopcellist I was having difficulties with the SCM_CREDENTIAL. When using various defaults "" was still being passed to the scm_credential which causes the play to look at every credential available and then subsequently fail.